### PR TITLE
Log Drains UI

### PR DESCRIPTION
**Pull request by jordienr**
URL: https://studio-staging-git-feat-log-drains-supabase.vercel.app/dashboard/project/_/settings/log-drains

- List log drains
- Create log drains (webhook, elastic and datadog)
- Update log drains
- Delete log drains

### CODE REVIEW

**Review by alaister — COMMENTED** [review]
Looking good! Everything appears to be working apart from descriptions, which aren't being saved from what I can tell

File: apps/studio/components/interfaces/LogDrains/LogDrainDestinationSheetForm.tsx [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      type: defaultType,
+      ...defaultValues,
+    },
+  })
+
+  const headers = form.watch('headers')
+
+  // IDK why but this useEffect is needed to set the default values
```
Comment (saltcod) [inline apps/studio/components/interfaces/LogDrains/LogDrainDestinationSheetForm.tsx:131]: this feels kinda weird. 
I haven't run into this but maybe Ivan / Joshen have

File: apps/studio/components/interfaces/LogDrains/LogDrainDestinationSheetForm.tsx [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+        </SheetHeader>
+        <SheetSection>
+          <Form_Shadcn_ {...form}>
+            <form
+              id={FORM_ID}
+              onSubmit={(e) => {
+                e.preventDefault()
+
+                const payload = form.getValues()
+                console.log('payload', payload)
```
Comment (saltcod) [inline apps/studio/components/interfaces/LogDrains/LogDrainDestinationSheetForm.tsx:206]: console.log

File: apps/studio/data/log-drains/update-log-drain-mutation.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  if (!payload.token) {
+    throw new Error('Token is required')
+  }
+
+  const { data, error } = await put('/platform/projects/{ref}/analytics/log-drains/{token}', {
+    params: { path: { ref: payload.projectRef, token: payload.token } },
+    body: {
+      name: payload.name,
+      description: payload.description,
+      config: payload.config as any,
```
Comment (saltcod) [inline apps/studio/data/log-drains/update-log-drain-mutation.ts:28]: got an easy type to add there?

File: apps/studio/hooks/misc/useCurrentOrgPlan.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { useOrgSubscriptionQuery } from 'data/subscriptions/org-subscription-query'
+import { useSelectedOrganization } from './useSelectedOrganization'
+
+export function useCurrentOrgPlan() {
```
Comment (saltcod) [inline apps/studio/hooks/misc/useCurrentOrgPlan.ts:4]: surprising this doesn't already exist!

**Review by saltcod — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -21,6 +21,7 @@
     "typecheck": "turbo --continue typecheck",
     "test:prettier": "prettier -c 'apps/**/*.{js,jsx,ts,tsx,css,md,mdx,json}'",
     "format": "prettier --write \"apps/**/*.{js,jsx,ts,tsx,css,md,mdx,json}\" \"packages/pg-meta/**/*.ts\"",
+    "format:studio": "prettier --write \"apps/studio/**/*.{js,jsx,ts,tsx,css,md,mdx,json}\"",
```
Comment (ivasilov) [inline package.json:24]: There's no need for this specific command. Just `format` is fast enough.

File: apps/studio/components/interfaces/LogDrains/LogDrainDestinationSheetForm.tsx [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+                    />
+                    <FormField_Shadcn_
+                      control={form.control}
+                      name="httpVersion"
+                      render={({ field }) => (
+                        <FormItem_Shadcn_>
+                          <FormControl_Shadcn_>
+                            <RadioGroupStacked
+                              onValueChange={field.onChange}
+                              defaultValue={field.value}
```
Comment (ivasilov) [inline apps/studio/components/interfaces/LogDrains/LogDrainDestinationSheetForm.tsx:271]: This should be `value`.

**Pull request merged**